src/httpProblemDetails.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpProblemDetailsDefinition } from './httpProblemDetailsDefinition';
+
+const invalidDetailName = ['type', 'status', 'title'];
+
+export class HttpProblemDetails {
+  public readonly type: string;
+  public readonly title: string;
+  public readonly status: number;
+  private additionalDetails: Map<string, any>;
```

Comment:
I'm not sure why the first three fields are public and the latter one is private; should they have the same level of privacy? I would say they are all private because the client code shouldn't read them other than via `toJson` method